aws/resource_aws_cloudfront_distribution.go

```diff
@@ -581,9 +581,33 @@ func resourceAwsCloudFrontDistribution() *schema.Resource {
 				Computed: true,
 			},
 			"active_trusted_signers": {
```

Comment:
It looks like the top level attribute name is what is really causing the unexpected behavior with core/SDK. For example, given this configuration:

```hcl
terraform {
  required_providers {
    aws = "2.70.0"
  }

  required_version = "0.12.29"
}

provider "aws" {
  region = "us-east-1"
}

resource "aws_cloudfront_distribution" "test" {
  # Faster acceptance testing
  enabled             = false
  retain_on_delete    = false
  wait_for_deployment = false

  default_cache_behavior {
    allowed_methods        = ["GET", "HEAD"]
    cached_methods         = ["GET", "HEAD"]
    target_origin_id       = "test"
    trusted_signers        = ["self"]
    viewer_protocol_policy = "allow-all"

    forwarded_values {
      query_string = false

      cookies {
        forward = "all"
      }
    }
  }

  origin {
    domain_name = "www.example.com"
    origin_id   = "test"

    custom_origin_config {
      http_port              = 80
      https_port             = 443
      origin_protocol_policy = "https-only"
      origin_ssl_protocols   = ["TLSv1.2"]
    }
  }

  restrictions {
    geo_restriction {
      restriction_type = "none"
    }
  }

  viewer_certificate {
    cloudfront_default_certificate = true
  }
}
```

Applying the above then overwriting the provider binary with this branch:

```console
# in the terraform-provider-aws directory
$ make build
# in the configuration directory
$ cp ~/go/bin/terraform-provider-aws .terraform/plugins/darwin_amd64/terraform-provider-aws_v2.70.0_x4; terraform init; terraform plan

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


Error: missing expected [
```

My guess is that the state handlers only understand the current attribute type during difference handling and ignore the fact its a read-only attribute that does not necessarily need to look at the old state for the attribute.

Reply:
ohhh interesting, so the subfield `items` name can actually be used but now just a matter of changing the root field's name? i'll give it a try on my end :) totally missed running an existing config locally to see how things behave 🤦‍♀️